packages/@aws-cdk-containers/ecs-service-extensions/lib/extensions/appmesh.ts

```diff
@@ -272,16 +272,30 @@ export class AppMeshExtension extends ServiceExtension {
       },
     });
 
+    var virtualRouterBaseListener: appmesh.VirtualRouterListener;
+    switch (this.protocol) {
+      case appmesh.Protocol.HTTP:
+        virtualRouterBaseListener = appmesh.VirtualRouterListener.http(containerextension.trafficPort);
+        break;
+      case appmesh.Protocol.HTTP2:
+        virtualRouterBaseListener = appmesh.VirtualRouterListener.http2(containerextension.trafficPort);
+        break;
+      case appmesh.Protocol.GRPC:
+        virtualRouterBaseListener = appmesh.VirtualRouterListener.grpc(containerextension.trafficPort);
+        break;
+      case appmesh.Protocol.TCP:
+        virtualRouterBaseListener= appmesh.VirtualRouterListener.tcp(containerextension.trafficPort);
+        break;
+      default:
+        throw new Error(`Protocol ${this.protocol} not supported by App Mesh Extension`);
+    }
```

Comment:
Can you instead refactor this to a private function or method, and use returns? Like this:

```ts
  private virtualRouterListener(): VirtualRouterListener {
    switch (this.protocol) {
      case appmesh.Protocol.HTTP: return appmesh.VirtualRouterListener.http(containerextension.trafficPort);
      case appmesh.Protocol.HTTP2: return appmesh.VirtualRouterListener.http2(containerextension.trafficPort);
      case appmesh.Protocol.GRPC: return appmesh.VirtualRouterListener.grpc(containerextension.trafficPort);
      case appmesh.Protocol.TCP: return appmesh.VirtualRouterListener. tcp(containerextension.trafficPort);
    }
  }
```

This way, it will be a compile-time (vs runtime) error when a new `Protocol` that is not handled here is added.